Plugins~/Src/MeshSyncClientBlender/python/unity_mesh_sync_common.py

```diff
@@ -102,15 +170,250 @@ def draw(self, context):
+    bl_idname = "meshsync.start_unity"
+    bl_label = "Export Objects"
+
+    def get_editor_version(self, directory):
+        project_version_path = directory + "/ProjectSettings/ProjectVersion.txt"
+        with open(project_version_path, "r+") as file:
+            first_line = file.readline()
+            version = first_line[len("m_EditorVersion: "):]
+            version = version[:-1]
+            return version
+
+    def launch_project(self, editor_path, project_path):
+        os = platform.system()
+        if os == 'Windows':
+            path = editor_path + " -projectPath \"" + project_path + "\""
+        elif os == 'Darwin':
+            path = editor_path + " -projectPath " + project_path
+        elif os == 'Linux':
+            path = editor_path + " -projectPath " + project_path
+
+        return subprocess.Popen(path)
+
+    def start_unity_project (self, directory):
+        
+        global unity_process
+
+        #Check if we have launched the project as a subprocess
+        if unity_process is not None and unity_process.poll() is None:
+            return 'ALREADY_STARTED'
+
+        #Check if there is an editor server listening from the target project
+        if msb_context.is_editor_server_available:
+            msb_context.sendEditorCommand(2)
+            reply_path = msb_context.editor_command_reply;
+            if reply_path == directory:
+                return 'ALREADY_STARTED'
+
+
+        editor_version = self.get_editor_version(directory)
+        editor_path = msb_get_editor_path(editor_version)
+
+        if not path.exists(editor_path):
+            return 'EDITOR_NOT_EXISTS'
+
+        #Launch the editor with the project
+        unity_process = self.launch_project(editor_path, directory)
+
+        return 'STARTED'
+
+
+    def execute(self, context):
+        #Try starting unity if not already started
+        directory = context.scene.meshsync_unity_project_path
+        start_status = self.start_unity_project(directory)
+
+        #If the project was launched now, wait until the Editor server is available
+        if start_status == 'STARTED':
+            while(msb_context.is_editor_server_available is False):
+                time.sleep(0.1)
+        elif start_status == 'EDITOR_NOT_EXISTS':
+            editor_version = self.get_editor_version(directory)
+            editor_path = msb_get_editor_path(editor_version)
+            message = "Editor path " + editor_path + " does not exist." 
+            MS_MessageBox(message = message)
+            print(message)
+            return {'FINISHED'}
+
+        bpy.ops.meshsync.create_server('INVOKE_DEFAULT')
+        return {'FINISHED'}
+
+class MESHSYNC_OT_CreateServer(bpy.types.Operator):
+    bl_idname = "meshsync.create_server"
+    bl_label = "Create a Scene Server if there is none"
+
+    def execute(self, context):
+       #If there is no server in the scene, add one
+        msb_context.sendEditorCommand(1)
```

Comment:
Is it possible to define a more meaningful variable for this 1 and 2 ?

Reply:
Definitely, will do

Reply:
done